### DISCUSSION TITLE
Euler angles from RGB values - EBSD tif file

### DISCUSSION THREAD

**Discussion by qdolle15** [pyxem/orix#444] [Q&A]
Dear all, 

I wonder if there is a way of getting back the Euler angles from a RGB array ? Kind of 'Orientation.from_euler()' but reversed ?

I'm trying to open an EBSD tif file and working with the crystal orientation but the tif file provides me only the RGB matrix. 
Have you any alternative way for doing that ? 

Thanks ! :)

**Comment by hakonanes** [comment]
Hi @qdolle15,

there is no such functionality in orix, I'm afraid.

Is your orientation color map colored according to a certain inverse pole figure? It is worth noting that you need a minimum of two such orientation maps to determine the *orientation* in each point. Orientations in an inverse pole figure map are colored according to which crystal *direction* is pointing in a sample direction. As such, in most materials, there are many different orientations which have a crystal direction in the same sample direction, say their c-axis out-of-plane but their a-axis pointing in different directions.

**Comment by qdolle15** [comment]
Thanks for your answer, 

You're right I miss this point but what if I have two IPF map according to different directions ?  It seems to be difficult... I will find an easier way

**Reply by hakonanes** [reply]
If you have two maps and know which directions they are colored with respect to, it should be possible.

There's another caveat... We only color orientations according to EDAX TSL's IPF in orix. So if the colors in your maps are produced with any other color key, we cannot do the inversion in orix.
